### ISSUE TITLE
Documentation about __file__ confused

### ISSUE THREAD

**Issue by mr-at-eo** [pyinstaller/pyinstaller#5626]
Documentation about `__file__` seems confused.

https://github.com/pyinstaller/pyinstaller/blame/develop/doc/runtime-information.rst#L100

>```python
>path_to_dat = (Path.cwd() / __file__).with_name("file.dat")
>```

in normal python, `__file__` is either relative to cwd or absolute. 
so one would use
```python
path_to_dat = Path(__file__).parent / "file.dat"
```

if  `z:/print_file.py` file contains

    print(__file__)

executing it from various places produces the following output:

    z:\>py z:/print_file.py
    z:/print_file.py
    
    z:\>py print_file.py
    print_file.py
    
    z:\>cd _temp
    
    z:\_temp>py z:/print_file.py
    z:/print_file.py
    
    z:\_temp>py ../print_file.py
    ../print_file.py

The obvious use is not possible after bundling, because the bootloader uses the relative path to itself instead of the one the python interpreter uses.
My followup questions:
1. Can the bootloader initiate `__file__` the same way that the python interpreter does? maybe behind a feature flag?
2. If this is not possible, should the documentation contain a paragraph about why not, to avoid someone else creating the same issue.

**Comment by bwoodsend** [comment]
What do you mean by *the obvious use*? 

> Can the bootloader initiate `__file__` the same way that the python interpreter does?

It's nothing to do with the bootloader. `__file__` is set by one of the runtime modules. But irregardless of that - what do you expect it to do differently? It can't point to your Python code because it's not a file. It's precompiled bytecode burried inside a zip archive inside of the executable. What it does is point to where it would be if code were just copy/pasted in so that data files and DLLs (which are just copy/pasted in) can be found without changing the code.

**Comment by mr-at-eo** [comment]
When executing a script with python, you can always use `__file__` to get a path to files sitting next to the script. no matter if they are absolute or relative, it just works.
After converting to an exe, it no longer just works.

    Z:\_temp>dist\print_file\print_file.exe
    print_file.py
    
I would have expected 

    dist\print_file\print_file.py

so that files that were added with `--add-data FILE;.` can be accessed.

Basically, it should take the path with which the exe was invoked into account.
Currently it only works well if I execute the exe from the directory it is in.

**Comment by bwoodsend** [comment]
> Z:\_temp>dist\print_file\print_file.exe
> print_file.py

You're right. That's not what it's supposed to do. It works fine if `print_file` is not the top level script. i.e. You have another file which does `import print_file` then run PyInstaller on that.
```
$ ./dist/main/main
/e/notebooks/pyi-issues/5626/dist/main/print_file.pyc
```
But `__file__` in the file being PyInstaller-ed directly just gives the name.
```
$ dist/print_file/print_file 
print_file.py
```
